src/rgw/store/sfs/types.h

```diff
@@ -88,10 +89,21 @@ class Bucket {
     const RGWBucketInfo &_bucket_info,
     const RGWUserInfo &_owner
   ) : cct(_cct), store(_store), name(_bucket_info.bucket.name),
-      bucket(_bucket_info.bucket), owner(_owner), creation_time(_bucket_info.creation_time) {
+      bucket(_bucket_info.bucket), owner(_owner),
+      creation_time(_bucket_info.creation_time),
+      placement_rule(_bucket_info.placement_rule) {
     _refresh_objects();
   }
 
+  RGWBucketInfo& to_RGWBucketInfo(RGWBucketInfo &out_info) const{
+    out_info.bucket = get_bucket();
+    out_info.owner = get_owner().user_id;
+    out_info.creation_time = get_creation_time();
+    out_info.placement_rule.name = placement_rule.name;
+    out_info.placement_rule.storage_class = placement_rule.storage_class; 
+    return out_info;
+  }
```

Comment:
I think this function should be called `to_rgw_bucket_info` or similar (`get_rgw_bucket_info` or `fill_rgw_bucket_info`). 
Name is not that important, but is to keep the same coding style. 

I'd need to check the `RGWBucketInfo` struct... but.. could be the case that returning it from a local variable is copy eluded and bond to a rvalue by the compiler? 
If that's case we would not need to pass the struct by reference and we could create it locally and just return. (As it would be moved)
Cleaner to me, as it won't require to declare the empty `RGWBucketInfo` before the call.

But again, we'd need to double check that and it's more a personal taste more than something that's wrong.

I did a quick test... adding a copy-constructor to the `RGWBucketInfo` struct and it's not called when doing:
```c++
auto test = bucketReg->to_RGWBucketInfo();
``` 
being `to_RGWBucketInfo`:
```c++
RGWBucketInfo to_RGWBucketInfo() const {
   RGWBucketInfo out_value;
   ... 
   ...
   return out_value;
```

Copy constructor is called if I do:
```c++
auto test = bucketReg->to_RGWBucketInfo(); // This doesn't call the copy constructor (copy elision -  move) 
auto test_copy = test; // This calls the copy constructor 
```

Reply:
I will modify `to_RGWBucketInfo` to `to_rgw_bucket_info`.

I understand your point here, I don't like too to declare an empty `RGWBucketInfo` on the stack before the call.
This design btw, to have an in/out `RGWBucketInfo` ref as parameter, is mainly to allow this method to be bound to an arbitrary caller's memory space.
Having `RGWBucketInfo` declared inside the method works in the same cases, but could be more inefficient because you are paying a copy when the destination memory already exists.
I agree that this particular usage case is equivalent because the destination memory does not exist yet at the time of the call (`info` is passed to a constructor so we should also provide an overloaded version of`to_rgw_bucket_info` returning an `rvalue` allowing it to be bound to a constructor accepting an `rvalue`).
In this form, btw, we can use this method as base for other overloaded versions (returning an `rvalue` for example).